receiver/hostmetricsreceiver/component/scraper.go

```diff
@@ -0,0 +1,69 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package component
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector/component"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/pdata"
+)
+
+// Scraper gathers metrics from the host machine and converts
+// these into internal metrics format.
+type Scraper interface {
+	// Initialize performs any timely initialization tasks such as
+	// setting up performance counters for initial collection.
+	Initialize() error
+	// Close cleans up any unmanaged resources such as performance
+	// counter handles.
+	Close() error
+	// ScrapeMetrics appends scraped metrics to the provided metric slice
+	ScrapeMetrics(ctx context.Context, metrics pdata.MetricSlice) error
+}
+
+// ScraperFactory can create a Scraper.
+type ScraperFactory interface {
+	component.Factory
```

Comment:
Another reason to use a different package for this. This was confusing about where the `Factory` is defined. Also why do you need to implement the `component.Factory`?

Reply:
Yea good point.

Initially I had been embedding a few things from `component` in these interfaces as the scrapers follow a fairly similar pattern to receivers in general. I decided to remove later, but missed this. I've changed ScraperFactory to scraper.Factory & declared the Type() function directly instead.